File: packages/ios/OpenCode/API/Models/FileEntry.swift
import Foundation

/// An entry from `GET /file` (directory listing) — used by the folder browser.
struct FileEntry: Decodable, Identifiable {
    let name: String
    let absolute: String
    let type: String          // "directory" | "file"
    let ignored: Bool?

    var id: String { absolute }
    var isDirectory: Bool { type == "directory" }
}

File: packages/android/app/src/main/java/studio/eugenezakharov/opencode/api/models/FileEntry.kt
package studio.eugenezakharov.opencode.api.models

import kotlinx.serialization.Serializable

/**
 * An entry from `GET /file` (directory listing) — used by the folder browser.
 * Mirrors the iOS `FileEntry`.
 */
@Serializable
data class FileEntry(
    val name: String,
    val path: String = "",
    val absolute: String,
    val type: String,          // "directory" | "file"
    val ignored: Boolean? = null,
) {
    val isDirectory: Boolean get() = type == "directory"
}
